fw/procfs.c

```diff
@@ -108,52 +108,63 @@ tfw_perfstat_collect(TfwPerfStat *stat)
 static int
 tfw_perfstat_seq_show(struct seq_file *seq, void *off)
 {
-#define SPRNE(m, e)	seq_printf(seq, m": %llu\n", e)
-#define SPRNED(m, e)	seq_printf(seq, m": %ums\n", e)
-#define SPRN(m, c)	seq_printf(seq, m": %llu\n", stat.c)
+#define SPRNE(m, e)  seq_printf(seq, m": %llu\n", e)
+#define SPRNED(m, e)  seq_printf(seq, m": %ums\n", e)
+#define SPRN(m, c)  seq_printf(seq, m": %llu\n", stat.c)
 
-	int i;
+	int i, ret;
 	TfwPerfStat stat = {0};
 	u64 serv_conn_active, serv_conn_sched;
 	SsStat *ss_stat = kmalloc(sizeof(SsStat) * num_online_cpus(),
-				  GFP_KERNEL);
+				GFP_KERNEL);
 	unsigned int val[T_PSZ] = { 0 };
 	TfwPrcntlStats pstats = {.val = val};
 
 	if (!ss_stat)
-		T_WARN("Cannot allocate sync sockets statistics\n");
+    	T_WARN("Cannot allocate sync sockets statistics\n");
 
 	if (health_stat_codes) {
 		stat.hm = kmalloc(tfw_hm_stats_size(health_stat_codes->ccnt),
-				  GFP_KERNEL);
+				GFP_KERNEL);
 		if (stat.hm)
 			tfw_hm_stats_clone(stat.hm, health_stat_codes);
 	}
 
 	tfw_perfstat_collect(&stat);
-	tfw_apm_stats_global(&pstats);
+	ret = tfw_apm_stats_global(&pstats);
+	if (ret < 0) {
+		seq_printf(seq, "Minimal response time\t\t: n/a\n");
+		seq_printf(seq, "Average response time\t\t: n/a\n");
+		seq_printf(seq, "Median  response time\t\t: n/a\n");
+		seq_printf(seq, "Percentiles\n");
+		for (i = TFW_PSTATS_IDX_ITH; i < ARRAY_SIZE(tfw_pstats_ith); ++i) {
+			seq_printf(seq, "%02d%%:\t n/a\n", tfw_pstats_ith[i]);
+		}
+		goto skip_apm;
+	}
 
 	SPRNED("Minimal response time\t\t", pstats.val[TFW_PSTATS_IDX_MIN]);
 	SPRNED("Average response time\t\t", pstats.val[TFW_PSTATS_IDX_AVG]);
 	SPRNED("Median  response time\t\t", pstats.val[TFW_PSTATS_IDX_P50]);
 	SPRNED("Maximum response time\t\t", pstats.val[TFW_PSTATS_IDX_MAX]);
 	seq_printf(seq, "Percentiles\n");
 	for (i = TFW_PSTATS_IDX_ITH; i < ARRAY_SIZE(tfw_pstats_ith); ++i)
-		seq_printf(seq, "%02d%%:\t%dms\n",
-				pstats.ith[i], pstats.val[i]);
+    	seq_printf(seq, "%02d%%:\t%dms\n",
+        	pstats.ith[i], pstats.val[i]);
+
+	skip_apm:
```

Comment:
```suggestion
skip_apm:
```
And other identation issues

Reply:
Still not fixed